.teamcity/src/subprojects/build/core/ProjectCore.kt

```diff
@@ -53,3 +62,8 @@ object ProjectCore : Project({
         }
     }
 })
+
+fun Requirements.defineOSAndMemoryAgentRequirements(os: String, memoryMb: Int) {
```

Comment:
Sorry, I should have clarified about the method name. The problems I see with this name:
* It's not clear how this memory amount will be used (as an upper limit, exact value, or something else).
* In the method name you just enumerate requirements you are defining. What if the third requirement will be needed in all those places?

The word `power` means different things for different people but it nicely conveys the intention. We could also think about a better word.

Reply:
What about medium to align with TC naming?

Reply:
But thinking about it I'm not convinced hardcoding the requirements into the function name. If tomorrow we change to low for instance, we'd have to refactor and have a bunch of changes reflected in many files for no real reason

Reply:
Yes, it's fine.

Reply:
The question is will you hypothetically update all places or only some?
If all of them then we can make two functions: one is `require` and call it like this: `require(os = "Linux", notLessMemory = 7000)`, another is requireMedium(os = "Linux") that will call `require` with predefined memory limit.